service/history/workflow/mutable_state_impl.go

```diff
@@ -4804,9 +4816,23 @@ func (ms *MutableStateImpl) AddWorkflowExecutionOptionsUpdatedEvent(
 		attachCompletionCallbacks,
 		links,
 	)
+	prevEffectiveVersioningBehavior := ms.GetEffectiveVersioningBehavior()
 	if err := ms.ApplyWorkflowExecutionOptionsUpdatedEvent(event); err != nil {
 		return nil, err
 	}
+
+	// Versioning Override set via UpdateWorkflowExecutionOptionsRequest
+	if versioningOverride != nil {
```

Comment:
the API may be called for other things too. let's only emit the metric if effective behavior OR effective version changes.